packages/web-components/src/calendar/calendar.ts

```diff
@@ -286,7 +286,28 @@ export class Calendar extends FASTCalendar {
    * @public
    */
   public handleSwitchMonth(month: number, year: number) {
```

Comment:
I think you could possibly reduce the number of calculations here and make it a bit more readable with something like..

```ts
public handleSwitchMonth(month: number, year: number) {
  this.selectedDates = '';

  // TODO: For range selection, instead of clearing out the selected dates,
  // filter them by current month

  const isPastYear = year < this.year;
  const isFutureYear = year > this.year;
  const isSameYearPastMonth = year === this.year && month < this.month;
  const isSameYearFutureMonth = year === this.year && month > this.month;

  // Check which transition to use for the primary panel
  if (isPastYear || isSameYearPastMonth) {
    this.prevMonthTransition();
  } else if (isFutureYear || isSameYearFutureMonth) {
    this.nextMonthTransition();
  }

  const yearPickerInfo = this.getYearPickerInfo();
  const isPastDecade = year < yearPickerInfo.decadeStart && this.yearPickerOpen;
  const isFutureDecade = year > yearPickerInfo.decadeEnd && this.yearPickerOpen;
  const isPastMonthPickerYear = year < this.monthPickerYear && !this.yearPickerOpen;
  const isFutureMonthPickerYear = year > this.monthPickerYear && !this.yearPickerOpen;

  // Check which transition to use for the secondary panel
  if (isPastDecade || isPastMonthPickerYear) {
    this.secondaryPanelTransition('previous');
  } else if (isFutureDecade || isFutureMonthPickerYear) {
    this.secondaryPanelTransition('next');
  }

  this.year = year;
  this.month = month;
  this.yearPickerDecade = year - (year % 10);
}
```

FYI  I didn't test this..